app/index.html

```diff
@@ -222,15 +219,58 @@
     window.cohortIdsList = ['hnsc_tcga'];
     window.iviz = {};
     window.iviz.datamanager = new DataManagerForIviz.init(window.cbioURL, studyCasesMap);
+    var getSelectableStudyIds = function() {
+      var def = new $.Deferred();
+      $.when(this.getPhysicalStudies(), this.getVirtualStudies()).then(function(physicalStudies, virtualStudies) {
+        var physicalStudyIds = _.pluck(physicalStudies,'studyId');
+        var virtualStudyIds = _.map(virtualStudies,function(virtualStudy){
+          if(virtualStudy['data'] && virtualStudy['data']['studies']){
+            return _.map(virtualStudy['data']['studies'],function(study){
+              return study['id'];
+            })
+          } else {
+            return [];
+          }
+        });
+        virtualStudyIds = _.flatten(virtualStudyIds);
+        def.resolve(virtualStudyIds)
+      }).fail(function(error) {
+        def.reject(error);
+      });
+      return def.promise();
+    }
+    var getVirtualStudies = function(){
```

Comment:
Just curious for future development, shouldn't this function access user email? Or the proxy endpoint will actually fetch studies for this user only?

Reply:
proxy api takes care of the email